openmetadata-ui/src/main/resources/ui/src/components/Settings/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx

```diff
@@ -100,81 +101,84 @@ const UserProfileDetails = ({
     [userData]
   );
 
-  const onDisplayNameChange = (e: React.ChangeEvent<HTMLInputElement>) =>
-    setDisplayName(e.target.value);
+  const onDisplayNameChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) =>
+      setDisplayName(isEmpty(e.target.value) ? undefined : e.target.value),
```

Comment:
@aniketkatkar97, I think we should include this `isEmpty` check when saving the displayName. This way, we can avoid calling `isEmpty` on the user's keystroke.